### PR TITLE
LensFlare: parameterize number of occlusion steps

### PR DESCRIPTION
**Pull request by scpeters**
Some testing of the LensFlare effects has indicated that the number of steps used when checking for occlusions in `LensFlareCompositorListener::OcclusionScale` is impacting performance. To investigate, I've exposed it as a parameter in the `rendering::LensFlare` c++ API and as an XML parameter `<occlusion_steps>` to the `LensFlarePlugin`.

### CODE REVIEW

**Comment by scpeters** [comment]
this is ready for review

File: gazebo/rendering/LensFlare.cc [hunk 1]
```diff
@@ -55,6 +55,10 @@ namespace gazebo
       /// \brief Color of lens flare.
       public: ignition::math::Vector3d color
           = ignition::math::Vector3d(1.0, 1.0, 1.0);
+
+      /// \brief Number of steps to take in each direction when checking for
+      /// occlusion.
+      public: double occlusionSteps = 10;
```
Comment (jacobperron) [inline gazebo/rendering/LensFlare.cc]:
```suggestion
      public: double occlusionSteps = 10.0;
```

File: gazebo/rendering/LensFlare.cc [hunk 2]
```diff
@@ -356,6 +367,9 @@ namespace gazebo
       public: ignition::math::Vector3d lensFlareColor
           = ignition::math::Vector3d(1.0, 1.0, 1.0);
 
+      /// \brief Color of lens flare.
+      public: double lensFlareOcclusionSteps = 10;
```
Comment (jacobperron) [inline gazebo/rendering/LensFlare.cc]:
```suggestion
      public: double lensFlareOcclusionSteps = 10.0;
```

File: plugins/LensFlareSensorPlugin.cc [hunk 3]
```diff
@@ -40,6 +40,9 @@ namespace gazebo
     public: ignition::math::Vector3d color
         = ignition::math::Vector3d(1.4, 1.2, 1.0);
 
+    /// \brief Lens flare occlusion steps
+    public: double occlusionSteps = 10;
```
Comment (jacobperron) [inline plugins/LensFlareSensorPlugin.cc]:
```suggestion
    public: double occlusionSteps = 10.0;
```

**Review by jacobperron — APPROVED** [review]
LGTM, only some minor comments

File: plugins/LensFlareSensorPlugin.cc (hunk 3 above)
Comment (scpeters) [reply plugins/LensFlareSensorPlugin.cc]: https://github.com/osrf/gazebo/pull/3234/commits/38fdb505b422d0a77356d17ac1cf7bfdf0883c47

File: gazebo/rendering/LensFlare.cc (hunk 2 above)
Comment (scpeters) [reply gazebo/rendering/LensFlare.cc]: https://github.com/osrf/gazebo/pull/3234/commits/38fdb505b422d0a77356d17ac1cf7bfdf0883c47

File: gazebo/rendering/LensFlare.cc (hunk 1 above)
Comment (scpeters) [reply gazebo/rendering/LensFlare.cc]: https://github.com/osrf/gazebo/pull/3234/commits/38fdb505b422d0a77356d17ac1cf7bfdf0883c47

**Review by jacobperron — APPROVED** [review]

**Pull request merged**